lib/full-site-editing/block-templates.php

```diff
@@ -46,15 +46,30 @@ function _gutenberg_get_template_file( $template_type, $slug ) {
 		get_stylesheet() => get_stylesheet_directory(),
 		get_template()   => get_template_directory(),
 	);
+
+	if ( 'wp_template_part' === $template_type ) {
+		$template_part_data = _gutenberg_get_template_part_info_from_theme_json();
+	}
+
 	foreach ( $themes as $theme_slug => $theme_dir ) {
 		$file_path = $theme_dir . '/' . $template_base_paths[ $template_type ] . '/' . $slug . '.html';
 		if ( file_exists( $file_path ) ) {
-			return array(
+
+			$new_template_item = array(
 				'slug'  => $slug,
 				'path'  => $file_path,
 				'theme' => $theme_slug,
 				'type'  => $template_type,
 			);
+
+			if ( 'wp_template_part' === $template_type ) {
+				return _gutenberg_conditionally_add_template_part_section_info(
+					$new_template_item,
+					$template_part_data
+				);
+			} else {
```

Comment:
I'm not sure what's the convention in the PHP part of Gutenberg, but I usually drop the `else` branch in this case and just leave the return on a newline. Since we are returning in the `if` branch we are safe to drop the `else` branch.

Reply:
Lol very true, im not sure what I was thinking when I put the `else` in.

Reply:
Yeah Marko pointed this out in [one of my PRs](https://github.com/WordPress/gutenberg/pull/28088#discussion_r557733884) as well. I think it's an unspoken convention to remove the return from an `else` branch (maybe worth making this an actual linting rule)

Reply:
I think not having the `else` makes the most sense and reads well.  I could swear I've seen a lint rule for that somewhere (maybe in JS or in another repo).  Anyways, updated!